components/automate-ui/src/app/pages/automate-settings/automate-settings.component.spec.ts

```diff
@@ -88,78 +130,230 @@ describe('AutomateSettingsComponent', () => {
+      // Client Runs
+      ['clientRunsRemoveNodes', 'converge-history',
+        genNestedIngestJob('infra', 'periodic_purge_timeseries', 'converge-history', 7, true)],
+
+      // Compliance
+      ['complianceRemoveReports', 'compliance-reports',
+        genNestedIngestJob('compliance', 'periodic_purge', 'compliance-reports', 8, true)],
+      ['complianceRemoveScans', 'compliance-scans',
+        genNestedIngestJob('compliance', 'periodic_purge', 'compliance-scans', 9, true)]
+    ], function (formName: string, nestedName: string, job: IngestJob) {
+        it(`when ${formName} form is saved as disabled, threshold is undefined
+            because it is not present in the nested form anymore`, () => {
+        const thisJobScheduler = new JobSchedulerStatus([job]);
+        component.updateForm(thisJobScheduler);
+
+        const newFormValues = component[formName].value;
+        const jobData = job.purge_policies.elasticsearch.find(item => item.name === nestedName);
+
+        expect(newFormValues.threshold).toEqual(undefined);
+        expect(newFormValues.disabled).toEqual(jobData.disabled);
+      });
     });
 
-    it('does not updates the "eventFeed" form group', () => {
-      component.updateForm(mockJobSchedulerStatus);
+    using([
+      // Client Runs
+      ['clientRunsRemoveData', genInjestJob('infra', 'missing_nodes', '5m', true)],
+      ['clientRunsLabelMissing', genInjestJob('infra', 'missing_nodes_for_deletion', '6h', true)]
+    ], function (formName: string, job: IngestJob) {
+      it(`when ${formName} form is saved as disabled, unit and threshold are undefined
+            because they are not present in the non-nested form anymore`, () => {
+        const thisJobScheduler = new JobSchedulerStatus([job]);
+        component.updateForm(thisJobScheduler);
 
-      const eventFeedValues = component.automateSettingsForm
-        .controls.eventFeed.value;
+        const newFormValues = component[formName].value;
 
-      // These are the defaults
-      expect(eventFeedValues.disable).toEqual(false);
-      expect(eventFeedValues.threshold).toEqual('');
-      expect(eventFeedValues.unit).toEqual('d');
+        expect(newFormValues.threshold).toEqual(undefined);
+        expect(newFormValues.unit).toEqual(undefined);
+        expect(newFormValues.disabled).toEqual(job.disabled);
+      });
     });
 
+
+
     describe('when user applyChanges()', () => {
       it('saves settings', () => {
         component.updateForm(mockJobSchedulerStatus);
         component.applyChanges();
-        expect(component.formChanged).toEqual(false);
+
+        // expect(component.notificationVisible).toBe(true);
         expect(component.notificationType).toEqual('info');
         expect(component.notificationMessage)
-          .toEqual('All settings have been updated successfully');
-        expect(component.notificationVisible).toEqual(true);
+        .toEqual('All settings have been updated successfully');
```

Comment:
sorry, shoulda commented on the test too

```suggestion
        .toEqual('Settings saved.');
```

Reply:
nah, that's not your responsibility - that was on me.  I pushed up and went to bed without checking the tests.  Sorry about that!